addons/sale_expense/models/sale_order.py

```diff
@@ -12,16 +12,22 @@ class SaleOrder(models.Model):
     expense_count = fields.Integer("# of Expenses", compute='_compute_expense_count', compute_sudo=True)
 
     @api.model
-    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
+    def _search_display_name(self, operator, value):
         """ For expense, we want to show all sales order but only their display_name (no ir.rule applied), this is the only way to do it. """
         if (
             self._context.get('sale_expense_all_order')
             and self.env.user.has_group('sales_team.group_sale_salesman')
             and not self.env.user.has_group('sales_team.group_sale_salesman_all_leads')
         ):
-            domain = expression.AND([domain or [], ['&', ('state', '=', 'sale'), ('company_id', 'in', self.env.companies.ids)]])
-            return super(SaleOrder, self.sudo())._name_search(name, domain, operator, limit, order)
-        return super()._name_search(name, domain, operator, limit, order)
+            if operator in expression.NEGATIVE_TERM_OPERATORS:
+                positive_operator = expression.TERM_OPERATORS_NEGATION[operator]
+            else:
+                positive_operator = operator
```

Comment:
Couldn't we factor out some function to simplify this `if else` being repeated over and over?
We could even think about applying the negation to several operators at once, since in this case we have:
* if `operator` is positive, we use `operator` in subdomains and `"in"` in the final domain
* if `operator` is negative, we use its negation in subdomains and `"not in"` in the final domain

I am thinking to some API like
```py
new_operator1, new_operator2, ... = to_positive(operator1, operator2, ...)
```
were the result is `(operator1, operator2, ...)` if `operator1` is positive, and the negation of all operators if `operator1` is negative.  In this method, we would use
```py
positive_operator, global_operator = to_positive(operator, 'in')
```

Not sure it's a good idea, since it makes quite some magic.  But it would factor out all the variants for positive and negative cases.  Note that we could also explicitly do
```py
if operator in expression.NEGATIVE_TERM_OPERATORS:
    positive_operator = expression.TERM_OPERATORS_NEGATION[operator]
    global_operator = 'not in'
else:
    positive_operator, global_operator = operator, 'in'
```